h2o-algos/src/test/java/hex/glm/GLMTweedieDispersionOnlyTest.java

```diff
@@ -0,0 +1,772 @@
+package hex.glm;
+
+import hex.DataInfo;
+import org.apache.commons.math3.special.Gamma;
+import org.junit.Assert;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import water.DKV;
+import water.Scope;
+import water.TestUtil;
+import water.fvec.Frame;
+import water.fvec.Vec;
+import water.runner.CloudSize;
+import water.runner.H2ORunner;
+import water.util.ArrayUtils;
+import water.util.Log;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.stream.IntStream;
+
+import static hex.glm.GLMModel.GLMParameters.DispersionMethod.pearson;
+import static hex.glm.GLMModel.GLMParameters.Family.tweedie;
+import static org.apache.commons.math3.special.Gamma.logGamma;
+
+@RunWith(H2ORunner.class)
+@CloudSize(1)
+public class GLMTweedieDispersionOnlyTest extends TestUtil {
+    
+    @Test
+    public void testTweedieDispersionPLess2() {
+        Scope.enter();
+        try {
+            // test with variance power > 1 and < 2
+            Frame train2 = parseAndTrackTestFile("smalldata/glm_test/tweedie_1p8Power_2Dispersion_5Col_10KRows.csv");
+            assertMLBetterThanPearson(train2, 2, 1.8, 1.5, "resp", 15);
+            assertMLBetterThanPearson(train2, 2, 1.8, 2.5, "resp", 8);
+        } finally {
+            Scope.exit();
+        }
+    }
+
+    @Test
+    public void testTweedieDispersionPExceed2p3phi0p5() {
+        Scope.enter();
+        try {
+            // test with variance power > 2
+            Frame train = parseAndTrackTestFile("smalldata/glm_test/tweedie_p3_phi0p5_10KRows.csv");
+            assertMLBetterThanPearson(train, 0.5, 3, 0.1, "x", 8);
+            assertMLBetterThanPearson(train, 0.5, 3, 1.5, "x", 8);
+        } finally {
+            Scope.exit();
+        }
+    }
+
+    @Test
+    public void testTweedieDispersionPExceed2P3phi1() {
+        Scope.enter();
+        try {
+            // test with variance power > 2
+            Frame train = parseAndTrackTestFile("smalldata/glm_test/tweedie_p3_phi1_10KRows.csv");
+            assertMLBetterThanPearson(train, 1, 3, 1.5, "x", 8);
+            assertMLBetterThanPearson(train, 1, 3, 0.5, "x", 8);
+        } finally {
+            Scope.exit();
+        }
+    }
+
+    @Test
+    public void testTweedieDispersionPExceed2p3phi1p5() {
```

Comment:
This test is failing 4 times out of 5 runs.
```java
java.lang.AssertionError
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertTrue(Assert.java:52)
	at hex.glm.GLMTweedieDispersionOnlyTest.assertMLBetterThanPearson(GLMTweedieDispersionOnlyTest.java:136)
	at hex.glm.GLMTweedieDispersionOnlyTest.testTweedieDispersionPExceed2p3phi1p5(GLMTweedieDispersionOnlyTest.java:76)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at water.runner.H2ORunnerBefores.evaluate(H2ORunnerBefores.java:30)
	at water.TestUtil$3$TimerStatement.evaluate(TestUtil.java:453)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at water.runner.H2ORunner.leaf(H2ORunner.java:92)
	at water.runner.H2ORunner.runChild(H2ORunner.java:61)
	at water.runner.H2ORunner.runChild(H2ORunner.java:27)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at water.runner.H2ORunnerAfters.evaluate(H2ORunnerAfters.java:32)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater$1.execute(IdeaTestRunner.java:38)
	at com.intellij.rt.execution.junit.TestsRepeater.repeat(TestsRepeater.java:11)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:35)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:235)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:54)
```

Reply:
Can you send me more logs like this:

<img width="1118" alt="image" src="https://user-images.githubusercontent.com/7231712/189776352-b9466302-ef93-471b-97ff-e85faa4a8372.png">

Reply:
I was not able to reproduce it.  Please make sure it gets the latest version.

Reply:
The newest version fixed all the java tests for me (I ran the tests 10 times without any failure). Thank you!